lib/awful/placement.lua

```diff
@@ -800,8 +801,18 @@ end
 -- Check if the proposed geometry fits the screen
 local function fit_in_bounding(obj, geo, args)
     local sgeo   = get_parent_geometry(obj, args)
+    for key, value in pairs(sgeo) do
+        if type(value) == "number" then
+            sgeo[key] = gmath.round(value)
+        end
+    end
     local region = cairo.Region.create_rectangle(cairo.RectangleInt(sgeo))
 
+    for key, value in pairs(geo) do
+        if type(value) == "number" then
+            geo[key] = gmath.round(value)
+        end
+    end
```

Comment:
This applies changes to the geo table we get from parameters. I think it's a bad practice.